server.rb

```diff
@@ -153,6 +159,10 @@ def initiate_check_run
 
           # Parse each offense to get details and location
           file['offenses'].each do |offense|
+            # Limit the number of annotations to 50
```

Comment:
To avoid having to update `max_annotations` in each iteration, what do you think about just taking a slice of the first 50 `offenses` outside of the loop? The following will work even if there are less than 50 offenses.

```ruby
# process first 50 offenses
file['offenses'][0, 50].each do |offense|
  # process offense ...
done
```

Reply:
I like this much better! 👍

Reply:
@swinton thinking about this more, will this work if there is more than one file. The loop iterates first through the files then through the offenses in each file. Will this allow 50 offenses per file?

Reply:
> this work if there is more than one file. The loop iterates first through the files then through the offenses in each file. Will this allow 50 offenses per file?

Ah, nope this won't work, I hadn't appreciated that there are many files with potentially many offenses. Good catch! ⚾️

Reply:
@swinton thanks for confirming. I think all the changes are in. ⚡️ I'm going to merge this. 🎉 ✨ 

I'll merge the quickstart guides on Monday after I've moved this repo to github-developer. I'm hoping for a pretty short turnaround to supporting more than 50 annotations.